### PR TITLE
ENH: update raw span index template

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*
#113 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json [hunk 1]
```diff
@@ -38,9 +49,27 @@
       "endTime": {
         "type": "date_nanos"
       },
-      "resource.attributes.service.name": {
-        "ignore_above": 1024,
+      "status": {
```
Comment (kowshikn) [inline situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json:52]: Cool, this works. I tried running `status.code>0` in the KQL and this works.

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json [hunk 2]
```diff
@@ -38,9 +49,27 @@
...
         "type": "keyword"
+      },
+      "durationInNanos": {
+        "type": "long"
+      },
+      "events": {
+        "type": "nested"
+      },
+      "links": {
+        "type": "nested"
```
Comment (kowshikn) [inline situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json:72]: Just asking for my clarification, this definition will fail if index request is anyone of the following,
1. events: "some string"
2. events: { "atrributes": { "key":"value"}}

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json [hunk 3]
```diff
@@ -3,12 +3,19 @@
     "date_detection": false,
```
Comment (kowshikn) [inline situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json:3]: events is actually array of objects which contains a field called `time`. Now should we enable this flag or define the mapping events with `time` as `date_time_nanos`

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json (hunk 2 above)
Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json:72]: Yeah. If the above 2 coexists, define `events` as `nested` will lead to error. I do not know if any field type will work for the above scenario.

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json (hunk 3 above)
Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json:3]: Agreed.
```
{
  "time": "2020-11-03T00:01:19.277324Z",
  "name": "",
  "attributes": {
    "method": "GET",
    "level": "info",
    "event": "HTTP request received",
    "url": "/customer?customer=392"
  },
  "droppedAttributesCount": 0
}
```

Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json:3]: Maybe also enable the flag as there might be other fields fit into datetime

Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json]: Fixed

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json (hunk 2 above)
Comment (kowshikn) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json]: So the above 2 examples are both invalid, is there way to force and allow only,
`events: [ { time : ".....",  attributes : { key : "value" } } ]`

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json (hunk 3 above)
Comment (kowshikn) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json]: I think we shouldn't enable the flag we are explicitly mapping them. I am sure date_detection is an unwanted cost. So lets disable detection as we will going forward explicitly map date_time.

File: situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json (hunk 2 above)
Comment (chenqi0805) [reply situp-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json]: Let me check path_match for nested

**Review by kowshikn — APPROVED** [review]
Lets keep events and links as nested for now, I think if we have to add mappings, lets do it in the next iteration whenever we use those fields.

**Review by yadavcbala — APPROVED** [review]

**Pull request merged**